framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/core/internal/SecuredBase.java

```diff
@@ -0,0 +1,261 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.rsp.server.core.internal;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+import org.jboss.tools.rsp.secure.crypto.CryptoException;
+import org.jboss.tools.rsp.secure.model.ISecureStorageProvider;
+import org.jboss.tools.rsp.server.ServerCoreActivator;
+import org.jboss.tools.rsp.server.spi.model.IServerModel;
+
+import com.google.gson.Gson;
+/**
+ * Helper class for storing runtime and server attributes.
+ */
+public abstract class SecuredBase extends Base {
+	private ISecureStorageProvider secureStorage = null;
+	
+	/**
+	 * Create a new object.
+	 * 
+	 * @param file
+	 */
+	public SecuredBase(File file, ISecureStorageProvider storage) {
+		this(file, null, storage);
+	}
+	
+	/**
+	 * Create a new object.
+	 * 
+	 * @param file
+	 * @param id
+	 * @param secure storage
+	 */
+	public SecuredBase(File file, String id, ISecureStorageProvider provider) {
+		super(file, id); 
+		secureStorage = provider;
+	}
+	
+	private boolean isSecureKey(String key) {
+		return key != null && key.startsWith(IServerModel.SECURE_ATTRIBUTE_PREFIX);
+	}
+
+	/**
+	 * Returns <code>true</code> if the attribute is currently set, and <code>false</code>
+	 * otherwise.
+	 * 
+	 * @param attributeName
+	 * @return <code>true</code> if the attribute is currently set, and <code>false</code>
+	 *    otherwise
+	 */
+	@Override
+	public boolean isAttributeSet(String attributeName) {
+		if( !isSecureKey(attributeName))
+			return super.isAttributeSet(attributeName);
+		if( secureStorage != null && secureStorage.getSecureStorage() != null) {
+			String securedNode = getSecuredKey();
+			try {
+				return secureStorage.getSecureStorage().propertyExists(securedNode, attributeName);
+			} catch(CryptoException ce) {
+				// TODO log
```

Comment:
Please implement logging for all `// TODO log`.

Reply:
We don't have a proper logging solution set up just yet. It is currently piecemeal and done plugin-by-plugin. This needs to be tackled as a large task with a unified response. The current pieces for logging are further up the chain and cannot be used by such a low-level plugin at this time. 

I will open a new issue for logging.